enterprise/server/cmd/ci_runner/main.go

```diff
@@ -1107,19 +1156,20 @@ func (ar *actionRunner) Run(ctx context.Context, ws *workspace) error {
 
 		// If we get an OOM or a Bazel internal error, copy debug outputs to the
 		// artifacts directory so they get uploaded as workflow artifacts.
-		artifactsDir := artifactsPathForCommand(ws, i)
-		namedSetID := filepath.Base(artifactsDir)
 		if exitCode == bazelOOMErrorExitCode || exitCode == bazelInternalErrorExitCode {
 			jvmOutPath := filepath.Join(ar.rootDir, outputBaseDirName, "server/jvm.out")
 			if err := os.Link(jvmOutPath, filepath.Join(artifactsDir, "jvm.out")); err != nil {
 				ar.reporter.Printf("%sfailed to preserve jvm.out: %s%s\n", ansiGray, err, ansiReset)
 			}
 		}
 		if exitCode == bazelOOMErrorExitCode {
-			// TODO(Maggie): Use invocation ID of failed bazel command
-			heapDumpPath := filepath.Join(ar.rootDir, outputBaseDirName, fmt.Sprintf("%d.heapdump.hprof", i))
-			if err := os.Link(heapDumpPath, filepath.Join(artifactsDir, "heapdump.hprof")); err != nil {
-				ar.reporter.Printf("%sfailed to preserve heapdump.hprof: %s%s\n", ansiGray, err, ansiReset)
+			bazelInvocationIDs := ar.reporter.childInvocations
+			if len(bazelInvocationIDs) > 0 {
+				lastInvocationID := bazelInvocationIDs[len(bazelInvocationIDs)-1]
+				heapDumpPath := filepath.Join(ar.rootDir, outputBaseDirName, fmt.Sprintf("%s.heapdump.hprof", lastInvocationID))
+				if err := os.Link(heapDumpPath, filepath.Join(artifactsDir, "heapdump.hprof")); err != nil {
+					ar.reporter.Printf("%sfailed to preserve heapdump.hprof: %s%s\n", ansiGray, err, ansiReset)
+				}
```

Comment:
nit: can we remove the condition `len(bazelInvocationIDs) > 0`? there could be an OOM before we get the child invocation event I think (?)